script.js

```diff
@@ -0,0 +1,524 @@
+                keyElement.classList.add('keyboard__key_wide');
+            } else if (key.keyCode === 'CapsLock') {
+                keyElement.classList.add('keyboard__key_less-wide');
+            } else if (key.keyCode === 'Space') {
+                keyElement.classList.add('keyboard__key_very-wide');
+            } else if (key.keyCode === 'langKey') {
+                keyElement.classList.add('keyboard__key_inner');
+            }
+            for (let keyLang in key) {
+                // Добавляем обертку для разных языков
+                if (keyLang !== 'keyCode') {
+                    const span = document.createElement('span');
+                    span.classList.add(keyLang);
+                    if (keyLang !== lang) {
+                        span.classList.add('hidden');
+                    }
+                    for (let prop in key[keyLang]) {
+                        // Добавляем обертку для разных регистров
+                        const propElement = document.createElement('span');
+                        propElement.classList.add(prop);
+                        if (prop !== capsLock) {
+                            propElement.classList.add('hidden');
+                        }
+                        propElement.innerText = key[keyLang][prop];
+                        span.appendChild(propElement);
+                    }
+                    keyElement.appendChild(span);
+                }
+            }
+            // Add events to keys
+            switch (key.keyCode) {
+                case 'langKey': {
+                    keyElement.addEventListener('click', () => {
+                        changeLanguage();
+                    });
+                    break;
+                }
+                case 'Backspace': {
+                    keyElement.addEventListener('click', () => {
+                        value = value.substring(0, value.length - 1);
+                        onInput();
+                    });
+                    break;
+                }
+                case 'Tab': {
+                    keyElement.addEventListener('click', () => {
+                        value = value + '   ';
+                        onInput();
+                    });
+                    break;
+                }
+                case 'Space': {
+                    keyElement.addEventListener('click', () => {
+                        value = value + ' ';
+                        onInput();
+                    });
+                    break;
+                }
+                case 'ShiftLeft': {
+                    keyElement.addEventListener('mousedown', () => {
+                        changeCase();
+                    });
+                    keyElement.addEventListener('mouseup', () => {
+                        changeCase();
+                    });
+                    break;
+                }
+                case 'ShiftRight': {
+                    keyElement.addEventListener('mousedown', () => {
```

Comment:
Code duplicates the statement above. 
case 'ShiftLeft':
case 'ShiftRight':
{
                    keyElement.addEventListener('mousedown', () => {
                        changeCase();
                    });
                    keyElement.addEventListener('mouseup', () => {
                        changeCase();
                    });
                    break;
                }